src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/ProjectSystem/RazorProjectInfoEndpointPublisher.TestAccessor.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.ProjectSystem;
+
+internal partial class RazorProjectInfoEndpointPublisher
+{
+    internal static TestAccessor GetTestAccessor(LSPRequestInvoker requestInvoker, IProjectSnapshotManager projectSnapshotManager)
+        => new(new RazorProjectInfoEndpointPublisher(requestInvoker, projectSnapshotManager, TimeSpan.FromMilliseconds(1)));
```

Comment:
FWIW, this still isn't the pattern. :smile:

This should be an instance member that takes no arguments like so:

```C#
internal TestAccessor GetTestAccessor() => new(this);
```

Unit tests should create an instance of `RazorProjectInfoEndpointPublisher` and call `GetTestAccessor` on the result. This is important because `RazorProjectInfoEndpointPublisher` and unit tests need to be able to dispose of it:

```C#
using var publisher = new RazorProjectInfoEndpointPublisher(requestInvoker, projectManager, TimeSpan.FromMilliseconds(1));
var publisherAccessor = publisher.GetTestAccessor();
```

Reply:
We could also make test accessor disposable and have it dispose the instance it's wrapping, no? Seemed like it simplified the test code. I don't have strong objections to making it _exactly_ like it is in other implementation if needed of course. Is this pattern documented somewhere for my reference?

Reply:
I swear there is a code action to create a test accessor, and I've used it before, but I must admit I just tried it and it wasn't showing up for me in whatever random file I had open.